src/languageserverinstance.jl

```diff
@@ -270,10 +270,13 @@ function Base.run(server::LanguageServerInstance)
         else
             Base.display_error(stderr, err, bt)
         end
+    finally
+        put!(server.combined_msg_queue, (type = :close,))
+        close(server.combined_msg_queue)
     end
 
     @async try
-        while true
+        while isopen(server.symbol_results_channel)
             msg = take!(server.symbol_results_channel)
```

Comment:
Same here, I think in general these kind of checks whether something is in the right state or not don't work well in async/parallel code because we should never really assume that things don't change between that check and the next thing we call. So we need to handle that situation in the next line (280) in any case, so we might as well _only_ handle it there.

Reply:
Right, these two `isopen` checks are basically unnecessary. `take!` will always either return or throw an exception if the underlying channel gets closed while `wait`ing for something to happen, so the important logic here is in the `finally` block, which closes the aggregate channel.